src/field.rs

```diff
@@ -76,7 +76,7 @@ pub trait Field:
         limbs
     }
 
-    fn from_canonical_u8_vec(u8_limbs: Vec<u8>) -> Result<Self> {
+    fn from_canonical_u8_vec(u8_limbs: Vec<u8>) -> Option<Self> {
```

Comment:
(Doesn't really matter, but Option seems slightly more natural to me when there's just one failure case and the caller is expected to understand it. Let me know if you prefer with Result though.)

Reply:
As you prefer, I tend to like `Result`s when there is a failure involve so that I can add a meaningful error message for future implementers.
But in this specific case,  an `Option` is perfectly fine since the code is very simple, and is more coherent with the rest of the code.

Reply:
Yeah that's a fair point. I'm really on the fence in this case, would be happy with either one.

Reply:
Also I think for complex functions (e.g. `generate_proof`) it might be cleaner to return a `Result` and use the `?` operator whenever an operation fails. If we use `Option`s, we'd have to `unwrap` all possibly failing operations or write a custom message in `except`, which is imo a bit less clean.

Reply:
Ah yeah that could be nice. Switching it back.